src/dali/plugin/pair_converter/historic_crypto.py

```diff
@@ -63,3 +85,112 @@ def get_historic_bar_from_native_source(self, timestamp: datetime, from_asset: s
                 retry_count += 1
 
         return result
+
+    # Monkey Patching for the HistoricalData class in the Historic_Crypto module
```

Comment:
> I don't think we should be patching dependent modules in the plugin: 
Me, neither.

> this looks like a good PR for Historic_Crypto (assuming they still accept PRs). 
Yes, it does. Unfortunately, it looks like the project has been abandoned. I emailed the owner and have gotten no response. I was thinking of just forking it and renaming it and applying this to it.

Another option is to simply absorb the module into this project.

> If they don't accept PRs I'm OK with deprecating the Historic_Crypto pair converter altogether and just use the other ones we have.
I think that would be possible, but the CCXT module probably needs to be updated in order to use the new API. This will most likely cause errors that we will need to troubleshoot and fix. I'm just looking for something that will work right now. The Coinbase Pro locked plugin does not work. You can use the default exchange, but that requires downloading the giant Kraken CSV which is also problematic. Right now, they released a new CSV file, but didn't update the link on their page, so it's dead.

The Historic Crypto module offers a simple solution that is adequate for a good majority of people who simply deal with the major crypto assets in USD. The CCXT pair converter is very powerful, but it breaks, a lot. I would really like to write a lot more tests for it before relying on it completely. We should have some kind of backup.

Reply:
OK, in that case I vote for forking/renaming Historic Crypto (without absorbing it into DaLI, to keep things modular) and then changing the dependency in DaLI to the new version of HC.

Reply:
Looking around, the official Coinbase Advanced Python SDK [allows access to public endpoints](https://github.com/coinbase/coinbase-advanced-py/blob/master/README.md#accessing-public-endpoints-without-authentication) without authentication. It is more aggressive with rate limiting, but it works out of the box. I can probably create a new plugin quickly that makes use of it. And if users do go through the work of creating API keys they can add them for faster pricing resolution.

Since the `historic_crypto` plugin is no longer necessary to fix the errors we are having I'm just going to revert the monkey patch so that we can get this PR finished.